url_matcher/matcher.py

```diff
@@ -0,0 +1,176 @@
+"""
+The matcher module contains the UrlMatcher class.
+"""
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Tuple, Union
+
+from url_matcher.patterns import PatternMatcher, get_pattern_domain, hierarchical_str
+from url_matcher.util import get_domain
+
+
+@dataclass
+class Patterns:
+    include: List[str]
+    exclude: List[str] = field(default_factory=list)
+    priority: int = 500
+
+    def get_domains(self) -> List[str]:
+        domains = [get_pattern_domain(pattern) for pattern in self.include]
+        return [domain for domain in domains if domain]
+
+    def get_includes_without_domain(self) -> List[str]:
+        return [pattern for pattern in self.include if get_pattern_domain(pattern) is None]
+
+    def all_includes_have_domain(self) -> bool:
+        """Return true if all the include patterns have a domain"""
+        return not self.get_includes_without_domain()
+
+    def is_universal_pattern(self) -> bool:
+        """Return true if there are no include patterns or they are empty. A universal pattern matches any domain"""
+        for pattern in self.include:
+            if pattern:
+                return False
+        return True
+
+    def get_includes_for(self, domain):
```

Comment:
Let's add an annotation for consistency:
```python
def get_includes_for(self, domain: str) -> List[str]:
```

Reply:
https://github.com/zytedata/url-matcher/pull/2/commits/f588f3ecd6b3f46fe932aff87c07872543282a9a